driver-ui-desktop/src/main/java/eu/tsystems/mms/tic/testframework/webdrivermanager/DesktopWebDriverFactory.java

```diff
@@ -248,33 +247,35 @@ public void setupSession(EventFiringWebDriver eventFiringWebDriver, DesktopWebDr
         }
     }
 
-    private void setWindowSizeBasedOnDisplayResolution(WebDriver.Window window, String browser) {
-        log().debug("Trying to set window size to: " + Defaults.DISPLAY_RESOLUTION);
-        String[] split = Defaults.DISPLAY_RESOLUTION.split("x");
+    private void setWindowSize(WebDriver.Window window, Dimension dimension) {
+        window.setPosition(new Point(0, 0));
+        window.setSize(dimension);
+    }
+
+    private Dimension getTargetWindowSize() {
+        String windowSizeProperty = PropertyManager.getProperty(TesterraProperties.WINDOW_SIZE, PropertyManager.getProperty(TesterraProperties.DISPLAY_RESOLUTION, "1920x1080"));
+        String[] split = windowSizeProperty.split("x");
         int width = Integer.parseInt(split[0]);
         int height = Integer.parseInt(split[1]);
+        return new Dimension(width, height);
+    }
+
+    private void setWindowSizeBasedOnDisplayResolution(WebDriver.Window window, String browser) {
+        Dimension dimension = getTargetWindowSize();
         try {
-            window.setSize(new Dimension(width, height));
-        } catch (Throwable t2) {
-            log().error("Could not set window size", t2);
+            setWindowSize(window, dimension);
+        } catch (Throwable t) {
+            log().error("Could not set window size", t);
 
             if (Browsers.edge.equals(browser)) {
```

Comment:
Why we have this check here? Is it for Legacy Edge? Maybe with Chromium Edge there should be no problem here.